src/codons.py

```diff
@@ -35,11 +37,24 @@ def split_codons(dna: str) -> list[str] | None:
     True
 
     """
-    # FIXME: Implement the function
-    return []
+    if dna == None:
+        output = None
```

Comment:
If you just return `None` here, you don't have to check if `dna != None` right after. You can often avoid deeply nested `if` blocks just by returning when you already know the answer.